audit.go

```diff
@@ -505,6 +500,49 @@ func (c *AuditClient) getReply(seq uint32) (*syscall.NetlinkMessage, error) {
 	return &msg, nil
 }
 
+// an isolated operation to unset our pid from the audit subsystem and close the socket
+func (c *AuditClient) closeAndUnsetPid() {
+	msg := syscall.NetlinkMessage{
+		Header: syscall.NlMsghdr{
+			Type:  AuditSet,
+			Flags: syscall.NLM_F_REQUEST | syscall.NLM_F_ACK,
+		},
+		Data: AuditStatus{
+			Mask: AuditStatusPID,
+			PID:  0,
+		}.toWireFormat(),
+	}
+
+	noParse := func(bytes []byte) ([]syscall.NetlinkMessage, error) {
+		return nil, nil
+	}
+
+	// If our request to unset the PID would block, then try to drain events from
+	// the netlink socket, resend, try again.
+	// In netlink, EAGAIN usually indicates our read buffer is full.
+	// The auditd code (which I'm using as a reference implementation) doesn't wait for a response when unsetting the  audit pid.
+	_, err := c.Netlink.SendNoWait(msg)
+	if err != nil {
+		// sending may be blocked, retry.
+		for {
```

Comment:
This is unbound, it should be guarded with a retry count, especially since there is a error check missing in the end.